web/src/components/AssertionForm/AssertionFormProvider.tsx

```diff
@@ -32,26 +33,28 @@ const initialFormProps = {
 };
 
 export const Context = createContext<ICreateAssertionModalProviderContext>({
-  isCollapsed: false,
-  setIsCollapsed: noop,
+  drawerState: 'INITIAL',
+  setDrawerState: noop,
   isOpen: false,
   open: noop,
   close: noop,
   formProps: initialFormProps,
   onSubmit: noop,
 });
 
-export const useAssertionForm = () => useContext(Context);
+export const useAssertionForm = () => useContext<ICreateAssertionModalProviderContext>(Context);
 
 const AssertionFormProvider: React.FC<{testId: string}> = ({children}) => {
   const dispatch = useAppDispatch();
-  const [isCollapsed, setIsCollapsed] = useState(true);
+  const [drawerState, setDrawerState] = useState<DrawerState>('INITIAL');
```

Comment:
I think more and more this is becoming the trace drawer provider instead of the assertion form provider, let's think about that for the future

Reply:
Agree. We can improve this later.